tasks/system_probe.py

```diff
@@ -1771,7 +1771,12 @@ def process_btfhub_archive(ctx, branch="main"):
                                             dst_file = os.path.join(dst_dir, file)
                                             if os.path.exists(dst_file):
                                                 raise Exit(message=f"{dst_file} already exists")
-                                            shutil.move(src_file, os.path.join(dst_dir, file))
+
+                                            shutil.move(src_file, dst_file)
+
+                                            # Always set the same mtime, so that the tarballs are deterministic
+                                            fixed_mtime = 0
+                                            os.utime(src_file, (fixed_mtime, fixed_mtime))
```

Comment:
We use `--mtime=@0` as an argument to `tar` in btfhub to accomplish this. Did you try that? It feels a little less brittle, since it is guaranteed. This method requires this call to be the last thing to touch the file. It also adds a syscall per file.